src/aggregate/aggregate_exec.c

```diff
@@ -213,39 +216,98 @@ static int buildRequest(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
   return rc;
 }
 
-static int execCommandCommon(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
-                             CommandType type) {
-  // Index name is argv[1]
-  if (argc < 2) {
-    return RedisModule_WrongArity(ctx);
-  }
+typedef struct searchConcurentCtx {
+  RedisModuleBlockedClient *bc;
+  RedisModuleString **argv;
+  int argc;
+  CommandType type;
+} searchConcurentCtx;
+
+static int executeSearchReplyCallback(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  return REDISMODULE_OK;
+}
+static int executeSearchTimeoutCallback(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  return REDISMODULE_OK;
+}
+
+static int execCommandInternal(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
+                               CommandType type) {
+  LockHandler_AcquireRead(ctx);
```

Comment:
Shouldn't you be acquiring the GIL before the read?

Reply:
It does not meter, LockHandler take care of it.